consumer.go

```diff
@@ -697,6 +672,34 @@ func (r *Consumer) startStopContinueBackoff(conn *Conn, success bool) {
 	}
 }
 
+func (r *Consumer) backoff() {
+	atomic.StoreInt64(&r.backoffDuration, 0)
+
+	if atomic.LoadInt32(&r.stopFlag) == 1 {
+		return
+	}
+
+	// pick a random connection to test the waters
+	conns := r.conns()
+	if len(conns) == 0 {
+		// backoff again
+		r.backoffMtx.RLock()
+		backoffDuration := r.backoffDurationForCount(r.backoffCounter)
+		r.backoffMtx.RUnlock()
+		atomic.StoreInt64(&r.backoffDuration, backoffDuration.Nanoseconds())
+		time.AfterFunc(backoffDuration, r.backoff)
```

Comment:
I don't think we need to wait for the entire (previously expired) backoff interval - I think we can just use some (small) constant sleep like `1s`.

Simplifies this inner block of code, too.